usr/src/uts/common/fs/zfs/zfs_vnops.c

```diff
@@ -4255,6 +4255,7 @@ zfs_putapage(vnode_t *vp, page_t *pp, u_offset_t *offp,
 		    &zp->z_pflags, 8);
 		zfs_tstamp_update_setup(zp, CONTENT_MODIFIED, mtime, ctime,
 		    B_TRUE);
+		(void)sa_bulk_update(zp->z_sa_hdl, bulk, count, tx);
```

Comment:
I am not confident that `sa_bulk_update` must always succeed, I _think_ that there could be cases where it could fail.
The reason I void-ed the return value is (1) for compatibility with the behavior before the change and (2) because the error would not be propagated to a "real caller".  This is `putpage` function and so the "real caller" is some code that modifies some memory.  Obviously it does not get any feedback from a failure to write the modified memory to its backing file. And in this case we are actually able to write the data, we just fail to update the file timestamps. In my opinion, it's better to treat that as a success.

I am not sure what illumos does if `putpage` returns an error.  FreeBSD would indefinitely keep the page in memory until the operation succeeds, if ever.
If illumos does the same, then maybe it's not that bad to return an error in the discussed scenario.   How do other illumos filesystems handle this?